src/common/tusb_verify.h

```diff
@@ -142,7 +146,9 @@
 #define ASSERT_1ARGS(_cond)            TU_VERIFY_DEFINE(_cond, _MESS_FAILED(); TU_BREAKPOINT(), false)
 #define ASSERT_2ARGS(_cond, _ret)      TU_VERIFY_DEFINE(_cond, _MESS_FAILED(); TU_BREAKPOINT(), _ret)
 
+#ifndef TU_ASSERT
```

Comment:
I am ok with this, however overwriting the TU_ASSERT() can be a bit of works, since it does have a macro trick to take optional parameter. If you has issue with TU_BREAKPOINT(), maybe you only need to overwrite it

Reply:
Not that big an issue, since I can reuse the tinyusb variadic helpers for that purpose, and redirect to [our assertion mechanism](https://modm.io/reference/module/modm-tinyusb/#debugging-tinyusb):

```c
// Redirect TinyUSB asserts to use modm_assert
#define MODM_ASSERT_1ARGS(_cond) \
	TU_VERIFY_DEFINE(_cond, modm_assert(0, "tu", \
		__FILE__ ":" MODM_STRINGIFY(__LINE__) " -> \"" #_cond "\""), false)

#define MODM_ASSERT_2ARGS(_cond, _ret) \
	TU_VERIFY_DEFINE(_cond, modm_assert_continue_fail(0, "tu", \
		__FILE__ ":" MODM_STRINGIFY(__LINE__) " -> \"" #_cond "\""), _ret)

#define TU_ASSERT(...) GET_3RD_ARG(__VA_ARGS__, MODM_ASSERT_2ARGS, MODM_ASSERT_1ARGS,UNUSED)(__VA_ARGS__)
```

This prints out:

```
Assertion 'tu' failed!
  modm/ext/tinyusb/portable/st/synopsys/dcd_synopsys.c:524 -> "epnum < 4U"
Abandoning...
```

which gets treated as an URI by my terminal, and when I ctrl-click on it, opens the file at the exact line in my editor. The TinyUSB assert formatter prints the function name and line number, which wasn't helpful for someone unfamiliar with the code. The above makes it relatively clear that you've run out of endpoints.